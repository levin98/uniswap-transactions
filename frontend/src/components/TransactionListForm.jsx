import React from "react";
import PropTypes from 'prop-types';
import { Button, DatePicker, Form, Input } from 'antd';
import 'antd/es/date-picker/style/index';
import 'antd/es/input/style/index';

const { RangePicker } = DatePicker;

const TransactionListForm = ({ onSubmit }) => {
    const [form] = Form.useForm();

    return (
        <Form form={form} layout='inline' onFinish={values => onSubmit(values)} data-testid="form">
            <Form.Item label="Transaction Hash" name="hash">
                <Input type="text" placeholder="Txn Hash" data-testid="form-hash" />
            </Form.Item>
            <Form.Item label="Time Range" name="timerange">
                <RangePicker data-testid="form-timerange" />
            </Form.Item>
            <Form.Item>
                <Button className="bg-[#282c34] text-white" htmlType="submit" data-testid="form-submit-button">Submit</Button>
            </Form.Item>
        </Form>
    );
}

TransactionListForm.propTypes = {
    onSubmit: PropTypes.func.isRequired
}

export default TransactionListForm;