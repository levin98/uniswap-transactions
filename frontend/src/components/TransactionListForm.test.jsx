import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import TransactionListForm from './TransactionListForm';
import "../mock"

test('renders the form', () => {
    const { container } = render(<TransactionListForm onSubmit={jest.fn()} />);

    expect(screen.getByTestId("form")).toBeInTheDocument();
    expect(screen.getByTestId("form-hash")).toBeInTheDocument();
    expect(screen.getByTestId("form-timerange")).toBeInTheDocument();
    expect(screen.getByTestId("form-submit-button")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});

test('calls onSubmit with the form values', async () => {
    const onSubmit = jest.fn();
    render(<TransactionListForm onSubmit={onSubmit} />);

    fireEvent.submit(screen.getByText("Submit"));
    await waitFor(() => expect(onSubmit).toHaveBeenCalledTimes(1));
});