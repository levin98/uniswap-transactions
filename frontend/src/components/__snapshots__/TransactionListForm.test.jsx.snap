// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the form 1`] = `
<div>
  <form
    class="ant-form ant-form-inline css-dev-only-do-not-override-k7429z"
    data-testid="form"
  >
    <div
      class="ant-form-item css-dev-only-do-not-override-k7429z"
    >
      <div
        class="ant-row ant-form-item-row css-dev-only-do-not-override-k7429z"
      >
        <div
          class="ant-col ant-form-item-label css-dev-only-do-not-override-k7429z"
        >
          <label
            class=""
            for="hash"
            title="Transaction Hash"
          >
            Transaction Hash
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control css-dev-only-do-not-override-k7429z"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input css-dev-only-do-not-override-k7429z"
                data-testid="form-hash"
                id="hash"
                placeholder="Txn Hash"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item css-dev-only-do-not-override-k7429z"
    >
      <div
        class="ant-row ant-form-item-row css-dev-only-do-not-override-k7429z"
      >
        <div
          class="ant-col ant-form-item-label css-dev-only-do-not-override-k7429z"
        >
          <label
            class=""
            for="timerange"
            title="Time Range"
          >
            Time Range
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control css-dev-only-do-not-override-k7429z"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="ant-picker ant-picker-range css-dev-only-do-not-override-k7429z"
                data-testid="form-timerange"
              >
                <div
                  class="ant-picker-input ant-picker-input-active"
                >
                  <input
                    autocomplete="off"
                    id="timerange"
                    placeholder="Start date"
                    readonly=""
                    size="12"
                    value=""
                  />
                </div>
                <div
                  class="ant-picker-range-separator"
                >
                  <span
                    aria-label="to"
                    class="ant-picker-separator"
                  >
                    <span
                      aria-label="swap-right"
                      class="anticon anticon-swap-right"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="swap-right"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="0 0 1024 1024"
                        width="1em"
                      >
                        <path
                          d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
                <div
                  class="ant-picker-input"
                >
                  <input
                    autocomplete="off"
                    placeholder="End date"
                    readonly=""
                    size="12"
                    value=""
                  />
                </div>
                <div
                  class="ant-picker-active-bar"
                  style="left: 0px; width: 0px; position: absolute;"
                />
                <span
                  class="ant-picker-suffix"
                >
                  <span
                    aria-label="calendar"
                    class="anticon anticon-calendar"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="calendar"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item css-dev-only-do-not-override-k7429z"
    >
      <div
        class="ant-row ant-form-item-row css-dev-only-do-not-override-k7429z"
      >
        <div
          class="ant-col ant-form-item-control css-dev-only-do-not-override-k7429z"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-k7429z ant-btn-default bg-[#282c34] text-white"
                data-testid="form-submit-button"
                type="submit"
              >
                <span>
                  Submit
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
